src/analytics/internal/analytics.js

```diff
@@ -0,0 +1,153 @@
+import { MapEventTypeToInt, MapNotifTypeToIntArray } from './constants'
+import SendToServer from './sendToServer'
+import { remoteFunction } from 'src/util/webextensionRPC'
+import AnalyticsLogStorage from './AnalyticsLogStorage'
+
+const getCount = remoteFunction('getCount')
+const getLatestEvent = remoteFunction('getLatestEvent')
+
+class Analytics {
+    _operationsMap = {
+        successful_search: undefined,
+        unsuccessful_search: undefined,
+        datepicker: undefined,
+        bookmark_filter: undefined,
+        tag_filter: undefined,
+        domain_filter: undefined,
+        tagging: undefined,
+        bookmark: undefined,
+        blacklist: undefined,
+        address_bar_search: undefined,
+        datepicker_nlp: undefined,
+        nlp_search: undefined,
+    }
+
+    async registerOperations() {
+        for (const event of Object.keys(this._operationsMap)) {
+            await this.loadInitialData(event)
+        }
+    }
+
+    async fromDexie(notifType) {
+        await this.loadInititalData(notifType, true)
+    }
+
+    async statisticsStorage(event) {
+        await this.incrementvalue(event)
+    }
+
+    /**
+     * Track any user-invoked events internally.
+     *
+     * @param {EventTrackInfo} eventArgs
+     */
+    async storeEventLogStatistics(eventArgs) {
+        const params = {
+            ...eventArgs,
+            type: MapEventTypeToInt[eventArgs.type].id,
+            details: eventArgs.details || {},
+        }
+
+        const notifParams = {
+            last_time_used: eventArgs.time,
+            notifType: MapEventTypeToInt[eventArgs.type].notifType,
+        }
+
+        await AnalyticsLogStorage(params)
+
+        if (MapEventTypeToInt[eventArgs.type].notifType) {
+            await this.statisticsStorage(notifParams)
+        }
+    }
+
+    /**
+     * Load Initial data from dexie for the particular event type
+     * Query to dexie and store into _operationsMap
+     * @param {notifType} type of notif event
+     */
+    async loadInitialData(notifType, isCacheUpdate = false) {
```

Comment:
`isCacheUpdate` is never set; it should be safe to remove

Reply:
It will be used in future, when we will work on `Activity-based notifications` then `write time` or `read time` we have to get the data from dexie itself for consistency.